autoload/db.vim

```diff
@@ -519,3 +602,129 @@ function! db#clobber_dbext(...) abort
   endfor
   return opts
 endfunction
+
+function! s:vim_job.cb(job, data) dict abort
+  if type(a:data) ==? type(0)
+    let self.exit = 1
+    let self.exit_status = a:data
+    return self.call_cb_if_finished()
+  endif
+  let self.output .= a:data
+endfunction
+
+function! s:vim_job.close_cb(channel) dict abort
+  let self.close = 1
+  return self.call_cb_if_finished()
+endfunction
+
+function! s:vim_job.call_cb_if_finished() abort
+  if self.close && self.exit
+    return self.callback(split(self.output, "\n", 1), self.exit_status)
+  endif
+endfunction
+
+function! s:nvim_job_cb(jobid, data, event) dict abort
+  if a:event ==? 'exit'
+    return self.callback(self.output, a:data)
+  endif
+  call extend(self.output, a:data)
+endfunction
+
+function! s:job_run(cmd, callback, stdin_file, ...) abort
+  let stdin = get(a:, 1, '')
+  if has('nvim')
+    let jobid = jobstart(a:cmd, {
+          \ 'on_stdout': function('s:nvim_job_cb'),
+          \ 'on_stderr': function('s:nvim_job_cb'),
+          \ 'on_exit': function('s:nvim_job_cb'),
+          \ 'output': [],
+          \ 'callback': a:callback,
+          \ 'stdout_buffered': 1,
+          \ 'stderr_buffered': 1,
+          \ })
+
+    if !empty(a:stdin_file) && filereadable(a:stdin_file)
+      let stdin = readfile(a:stdin_file, 'b')
+    endif
+
+    if !empty(stdin)
+      call chansend(jobid, stdin)
+      call chanclose(jobid, 'stdin')
+    endif
+
+    return jobid
+  endif
+
+  if exists('*job_start')
+    let fn = copy(s:vim_job)
+    let fn.callback = a:callback
+    let opts = {
+          \ 'out_cb': fn.cb,
+          \ 'err_cb': fn.cb,
```

Comment:
These can both be handled by the `'callback'` key. And if you rename `fn.cb` to `fn.callback` things get a whole lot easier to intuit.

Reply:
You mean by [`callback` ](https://vimhelp.org/channel.txt.html#job-options:~:text=job%2Dcallback%0A%22-,callback,-%22%3A%20handler) job options? Was that available in the older vim versions (8.0+) ?

I used `cb` instead of `callback` because `callback` is the actual callback that is passed as argument. We could rename that one to something else, and rename `cb` to `callback`.

Reply:
I switched to `callback` and renamed all `cb` to `callback` so it's more readable.

Reply:
Yeah, `"callback"` has been there since the beginning.